docs/powerpoint/insert-slides-into-presentation.md

```diff
@@ -0,0 +1,183 @@
+
+```javascript
+async function insertAllSlides() {
+  await PowerPoint.run(async function(context) {
+    context.presentation.insertSlidesFromBase64(chosenFileBase64);
+    await context.sync();
+  });
+}
+```
+
+You can control some aspects of the insertion result, including where the slides are inserted and whether they get the source or target formatting (and more), by passing an [InsertSlideOptions](/javascript/api/powerpoint/powerpoint.insertslideoptions) object as a second parameter to `insertSlidesFromBase64`. The following is an example. About this code, note:
+
+- There are two possible values for the `formatting` property: "UseDestinationTheme" and "KeepSourceFormatting". Optionally, you can use the full name of the `InsertSlideFormatting` enum, such as `PowerPoint.InsertSlideFormatting.useDestinationTheme`.
+- The function will insert the slides from the source presentation immediately after the slide specified by the `targetSlideId` property. The value of this property is a string of one of three possible forms: ***nnn*#**, **#*mmmmmmmmm***, or ***nnn*#*mmmmmmmmm***, where *nnn* is the slide's ID (typically 3 digits) and *mmmmmmmmm* is the slide's creation ID (typically 9 digits). Some examples are `267#763315295`, `267#`, and `#763315295`.
+
+```javascript
+async function insertSlidesDestinationFormatting() {
+  await PowerPoint.run(async function(context) {
+    context.presentation
+    .insertSlidesFromBase64(chosenFileBase64,
+                            {
+                                formatting: "UseDestinationTheme",
+                                targetSlideId: "267#"
+                            }
+                          );
+    await context.sync();
+  });
+}
+```
+
+Of course, you typically won't know at coding time the ID or creation ID of the target slide. More commonly, an add-in will ask users to select the target slide. The following steps show how to get the ***nnn*#** ID of the currently selected slide and use it as the target slide.
+
+1. Create a function that gets the ID of the currently selected slide by using the [Office.context.document.getSelectedDataAsync](/javascript/api/office/office.document#getSelectedDataAsync_coercionType__callback_) method of the Common JavaScript APIs. The following is an example. Note that the call to `getSelectedDataAsync` is embedded in a Promise-returning function. It is a good practice to do this because the Promise-returning function is easy to call (and await) inside the `run()` method of an application-specific API. By awaiting the Promise-returning function, you are assured that the callback function of the Common API method has completed before the next line in the `run()` method executes.
```

Comment:
Should we link to this documentation about this pattern? https://docs.microsoft.com/en-us/office/dev/add-ins/develop/asynchronous-programming-in-office-add-ins?view=powerpoint-js-preview#wrap-common-apis-in-promise-returning-functions
That would free this section for having to explain the purpose of Promise wrapping.